src/Illuminate/Foundation/Application.php

```diff
@@ -257,7 +257,7 @@ public static function inferBasePath()
             isset($_ENV['APP_BASE_PATH']) => $_ENV['APP_BASE_PATH'],
             default => dirname(array_values(array_filter(
                 array_keys(ClassLoader::getRegisteredLoaders()),
-                fn ($path) => ! str_contains($path, '/vendor/'),
+                fn ($path) => ! str_starts_with($path, 'phar://'),
```

Comment:
Checking for `phar://` would solve the original bug https://github.com/laravel/framework/issues/54118 and still allows `vendor`